models/filtermodel.h

```diff
@@ -27,18 +28,37 @@ class FilterModel : public QSortFilterProxyModel
 {
     Q_OBJECT
     Q_PROPERTY(DocumentListModel *sourceModel READ sourceModel WRITE setSourceModel NOTIFY sourceModelChanged)
+    Q_PROPERTY(SortParameter sortParameter READ sortParameter WRITE setSortParameter NOTIFY sortParameterChanged)
 public:
     FilterModel(QObject *parent = 0);
     ~FilterModel();
+    
+    enum SortParameter {
+        None,
+        Name,
+        Type,
+        Date
+    };
+
+    Q_ENUM(SortParameter)
 
 public:
     DocumentListModel* sourceModel() const;
+    
+    SortParameter sortParameter() const;
+    void setSortParameter(const SortParameter &sortParameter);
 
 public Q_SLOTS:
     void setSourceModel(DocumentListModel *model);
+    
+    void sortByParameter();
```

Comment:
Ends up similar, but this doesn't get used as slot, rather as Q_INVOKABLE. Then again is there reason to even have the method instead of just sorting when sort type is set?

Reply:
Yes, I agree, that it is used mostly as Q_INVOKABLE and it is not correct to use it as a slot, but what do you mean under "is there reason to even have the method instead of just sorting when sort type is set"? I need it to match sort role and  call sort() of QSortFilterProxyModel to sort items

Reply:
I mean now you have an api that you can use someModel.sortParameter = FilterModel.Date, but which won't be sorted according to time until the method here is explicitly called. So instead the sort mode setter could already trigger the new sorting.

Reply:
Done